073-decouple-ca-management-from-end-entity-certificate-management.md

```diff
@@ -0,0 +1,250 @@
+* <CLUSTER_NAME>-kafka-exporter-certs for the Kafka Exporter
+* <CLUSTER_NAME>-cruise-control-certs for Cruise Control
+
+Strimzi also uses the following annotations on Kubernetes Secrets:
+* strimzi.io/ca-cert-generation to indicate the generation of the Cluster CA or Client CA certificate in the Kubernetes Secret
+* strimzi.io/ca-key-generation to indicate the generation of the Cluster CA or Client CA private key in the Kubernetes Secret
+
+Strimzi uses the following annotations on Kubernetes Pods and Kubernetes Secrets to indicate the generation of the Cluster CA certificate used to sign the certificates that are either stored in the Kubernetes Secret, or presented by the Kubernetes Pod:
+* strimzi.io/cluster-ca-cert-generation for certificates signed by the Cluster CA
+* strimzi.io/clients-ca-cert-generation for certificates signed by the Clients CA
+
+Strimzi uses the following annotation on Kubernetes Pods to indicate the generation of the Cluster CA private key used to sign the cluster CA certificate that the pod currently trusts
+* strimzi.io/cluster-ca-key-generation
+
+### Existing reconcile flow
+
+The reconcile loop for certificates is split over a couple of classes that run one after another, those are:
+1. CaReconciler
+2. ZooKeeperReconciler
+3. KafkaReconciler
+4. EntityOperatorReconciler
+5. CruiseControlReconciler
+6. KafkaExporterReconciler
+
+Currently, as part of a reconciliation loop, the CaReconciler handles three things:
+1. Verifying the Cluster CA and Client CA Kubernetes Secrets. For each CA:
+   1. If Strimzi is managing the CA it either creates or renews the CA certificate and key and stores them in the Kubernetes Secrets listed above
+   2. If Strimzi is **not** managing the CA it checks that the Kubernetes Secrets for the CA certificate and private key are present in the cluster
+2. Reconciles the <CLUSTER_NAME>-cluster-operator-certs Kubernetes Secret used by the cluster operator's Admin client when connecting to Kafka and ZooKeeper
+   1. This only occurs if all pods in the cluster have the correct strimzi.io/cluster-ca-key-generation annotation, meaning the Cluster CA certificate that is being put into the Kubernetes Secret is already trusted by all components
+3. Maybe triggering a rolling update of the ZooKeeper and Kafka nodes to trust a new Cluster CA certificate. This only happens if either:
+   1. The Cluster CA key got replaced during the current reconciliation, or
+   2. Some pods do not trust the latest Cluster CA certificate, which implies the operator previously updated the Cluster CA key and a rolling update was started during a previous reconciliation, but it was not completed
+
+ZooKeeperReconciler and KafkaReconciler classes:
+1. Generate the certificates they will present to clients and store them in the Kubernetes Secrets listed above, with the cluster-ca-cert-generation (and clients-ca-cert-generation for Kafka) annotation set to the current value passed in with the Cluster CA (and Clients CA).
```

Comment:
re-reading again your subjects are ZooKeeperReconciler and KafkaReconciler here ... but then you say "they will present" ... it's not correct, because it's not the reconcilers presenting the certs but the ZooKeeper and Kafka nodes, so I would make it clearer first.
Regarding the "clients" issues maybe just adding ".... to clients (other ZooKeeper or Kafka nodes, or Kafka consumer or producer)"? Or maybe I am just overthinking :-D